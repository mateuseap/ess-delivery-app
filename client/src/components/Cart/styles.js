import styled from "styled-components";

import { StyledButton } from "../../utils/styles";
import { DEFAULT_RED } from "../App/App.style";

export const CartStyle = styled.div`
  width: 90%;
  margin: auto;
`;

export const TextStyle = styled.h1`
  text-align: center;
  font-weight: bold;
  padding-top: 1.2rem;
  color: black;
`;

export const RestaurantText = styled.div`
  font-size: 2rem;
  padding-left: 10%;
  line-height: 94px;
  padding-top: 0.5rem;
  color: ${DEFAULT_RED};
`;

export const HeaderRow = styled.div`
  font-size: 2rem;
  text-align: center;
  font-weight: bold;
  color: black;
  padding-top: 0.5rem;
  display: flex;
  flex-direction: row;
`;

export const ItemRow = styled.div`
  display: flex;
  flex-direction: row;
  font-size: 1.5rem;
  text-align: center;
  font-weight: bold;
  color: black;
  padding-top: 0.5rem;
  align-items: center;
`;

export const ItemNameHeader = styled.div`
  width: 30%;
  margin-left: 7vw;
`;
export const ItemName = styled.div`
  width: 30%;
`;

export const ItemPrice = styled.div`
  width: 20%;
`;
export const ItemQuantity = styled.div`
  width: 15%;
`;

export const ItemTotal = styled.div`
  margin-left: auto;
  text-align: end;
  width: 15%;
`;

export const AddButton = styled(StyledButton)`
  padding: 0;
  margin-left: 2%;
  margin-right: 2.5%;
  font-size: 2.5rem;
  width: 3.5rem;
  border-radius: 50%;
`;
export const RemoveButton = styled(StyledButton)`
  font-size: 3.4rem;
  padding: 0 7px 6px 5px;
  line-height: 1;
  width: 3.5rem;
  border-radius: 50%;
`;

export const OrderTotalStyle = styled.div`
  color: ${DEFAULT_RED};
  font-size: 1.5rem;
  margin-left: auto;
  width: fit-content;
  padding: 12px 5% 0 50px;
  font-weight: bold;
`;

export const DeliveryFeeNotice = styled.div`
  color: ${DEFAULT_RED};
  font-size: 0.75rem;
  margin-left: auto;
  width: fit-content;
  padding: 5px 5% 20px 50px;
  font-weight: bold;
`;

export const OrderButton = styled(StyledButton)`
  margin-left: auto;
  display: flex;
`;

export const ItemImg = styled.div`
  content: url(${(props) => props.src});
  border-radius: 50px;
  width: 7.5vw;
  height: 8.8vh;
`;

export const RedirectHomeButton = styled(StyledButton)`
  margin: 2% auto 0 auto;
`;
