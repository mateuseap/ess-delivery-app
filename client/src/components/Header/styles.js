import styled from "styled-components";

export const HeaderStyle = styled.div`
  margin-top: 1%;
  background: #91091e;
  display: flex;
  flex-direction: row;
  align-items: center;
  border-radius: 100px;
`;

export const Title = styled.div`
  color: white;
  font-size: 3em;
  line-height: 70px;
  left: auto;
  margin-left: 2%;
`;
export const Buttons = styled.div`
  display: flex;
  margin: 0.5% 1% 0.5% auto;
  * {
    margin-top: auto;
    margin-bottom: auto;
    padding-left: 12px;
    padding-right: 12px;
  }
`;

export const UserBg = styled.div`
  display: flex;
  width: 115px;
  height: 115px;
  background: #fffbfb;
  border-radius: 100px;
  align-items: center;
  padding: 0;
`;

export const UserPhoto = styled.div`
  content: url(${(props) => props.src});
  margin: auto;
  width: 110px;
  border-radius: 50%;
`;

export const CartItemCount = styled.div`
  position: absolute;
  right: 2px;
  top: -15px;
  color: white;
  padding: 0 8px;
  background-color: #da723c;
  border-radius: 50px;
`;
