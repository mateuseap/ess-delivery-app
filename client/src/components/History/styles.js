import styled from "styled-components";

export const ImageStyle = styled.img`
  width: 234px;
  height: 170px;
  border-radius: 65px;
  content: url(${(props) => props.photoUrl});
`;

export const DisabledSendButton = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
`;

export const DescriptionStyle = styled.div`
  display: grid;
  width: 200px;
  grid-auto-flow: row;
  grid-column-gap: 4vw;
`;

export const ActionButtonsStyle = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-column-gap: 15vw;
`;

export const RateLabel = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-column-gap: 5em;
  min-width: 30vw;
  font-size: 40px;
  line-height: 47px;
`;

export const MainDiv = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-column-gap: 10vw;
`;

export const TableBodyStyle = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const PageStyle = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const BorderText = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 30vw;
  color: white;
  line-height: 70px;
  font-weight: 400;
  text-align: center;
  background: rgba(250, 16, 52, 0.82);
  border-radius: 63px;
`;

export const RectangleFilter = styled.div`
  width: 230px;
  height: 60px;
  display: flex;
  gap: 10%;
  padding: 12px;

  color: white;
  font-weight: 400;
  font-size: 16px;
  line-height: 38px;

  justify-content: center;
  align-items: center;

  background: #91091e;
  border-radius: 100px;
`;

export const RectangleDaysFilter = styled.div`
  width: 80px;
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;

  background: rgba(250, 16, 52, 0.82);
  border-radius: 63px;
`;

export const SelectStyle = styled.select`
  background: transparent;
  line-height: 1;
  border: 0;
  color: white;
  border-radius: 63px;
  text-align: center;
  height: 100%;
  width: 100%;
  -webkit-appearance: none;
`;

export const TopDiv = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-column-gap: 10vw;
`;

export const ButtonStyle = styled.button`
  all: unset;
  cursor: pointer;
  width: 40px;
  height: 40px;
  text-align: center;
  background: rgba(250, 16, 52, 0.82);
  border-radius: 63px;
  box-sizing: border-box;
  &:hover {
    border: 1px solid yellow;
    background-color: rgba(250, 16, 52, 0.2);
  }
`;

export const NoDataStyle = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  top: 45%;
  left: 0;
  right: 0;
  margin: auto;
`;

export const OptionStyle = styled.option`
  color: black;
`;
