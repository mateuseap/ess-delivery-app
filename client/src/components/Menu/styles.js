import styled from "styled-components";

export const PageStyle = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const TableStyle = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  gap: 10%;
`;

export const TableBodyStyle = styled.div`
  background-color: #ffffff;
  margin: 15px 0;
  border-radius: 65px;
  display: grid;
  grid-auto-flow: column;
  height: 285px;
  place-items: center;
`;

export const ItemPhoto = styled.img`
  content: url(${(props) => props.photo});
  height: 215px;
  width: 215px;
  border-radius: 65px;
`;

export const ItemData = styled.div`
  display: flex;
  flex-direction: column;
`;

export const ItemDescriptionStyle = styled.p`
  font-size: 18px;
  font-weight: 400;
  max-width: 350px;
`;

export const RatingTextStyles = styled.h4`
  color: #383838;
  font-weight: 600;
  font-size: 2rem;
`;
