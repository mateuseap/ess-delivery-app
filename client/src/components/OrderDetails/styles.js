import styled from "styled-components";

export const OrderDetailsPage = styled.div`
  padding: 10px;
  flex-direction: column;
  align-items: center;
`;

export const OrderTitles = styled.h1`
  font-weight: bold;
  color: #91091e;
  text-align: center;
  margin-top: 20px; 
`;

export const OrderStatus = styled.div`
  align-items: center;
  height: 170px;
  padding-top: 0.5%;
  display: flex;
  flex-direction: column;
`;

export const OrderStatusItem = styled.div`
  text-align: center;
  color: white;
  flex-direction: row;
  background-color: #91091e;
  border-radius: 100px;
  width: 20vw;
  font-weight: bold;
  margin: 3px;
  padding: 5px;

`;

export const Restaurant = styled.h4`
  font-weight: bold;
  color: #91091e;
  display: flex;
  align-items: left;
`;

export const OrderItem = styled.div`
  height: 30px;
  font-size: 20px;
  position: absolute;
  font-weight: bold;
`;

export const OrderItemPrice = styled.div`
  height: 30px;
  font-size: 20px;
  position: relative;
  text-align: right;
  font-weight: bold;
`;

export const TotalPrice = styled.div`
  height: 30px;
  font-size: 25px;
  color: #91091e;
  position: relative;
  text-align: right;
  font-weight: bold;
`;

export const Deliver = styled.div`
  margin-top: 3%;
  height: 30px;
  color: #91091e;
  font-size: 20px;
  position: absolute;
  font-weight: bold;
  margin-bottom: 2%;
`;

export const ButtonRight = styled.div`
  margin-top: 3%;
  height: 30px;
  font-size: 25px;
  color: #91091e;
  position: relative;
  text-align: right;
  font-weight: bold;
  margin-bottom: 2%;
`;
