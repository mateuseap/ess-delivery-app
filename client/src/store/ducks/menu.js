import { createReducer, createActions } from "reduxsauce";
import Immutable from "seamless-immutable";

export const { Types, Creators } = createActions({
  menuRequest: [],
  getMenu: ["id"],
  menuSuccess: ["data"],
  menuError: ["err"],
});

export const INITIAL_STATE = Immutable({
  data: [],
  loading: false,
  error: false,
});

const request = (state) => state.merge({ loading: true });

const error = (state, { err }) => state.merge({ loading: false, error: err });

const success = (state, { data }) =>
  state.merge({ data, loading: false, error: false });

export default createReducer(INITIAL_STATE, {
  [Types.MENU_ERROR]: error,
  [Types.MENU_REQUEST]: request,
  [Types.MENU_SUCCESS]: success,
});
