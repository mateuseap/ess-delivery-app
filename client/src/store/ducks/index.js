import { combineReducers } from "redux";

import { reducer as toastr } from "react-redux-toastr";

import user from "./user";
import restaurants from "./restaurants";
import history from "./history";
import cart from "./cart";
import menu from "./menu";
import order from "./order";

export default combineReducers({
  user,
  restaurants,
  history,
  cart,
  menu,
  order,
  toastr,
});
