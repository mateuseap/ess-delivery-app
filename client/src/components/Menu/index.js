import React, { Component } from "react";
import { connect } from "react-redux";

import { Creators as MenuCreators } from "../../store/ducks/menu";
import { Creators as CartCreators } from "../../store/ducks/cart";

import {
  ItemPhoto,
  PageStyle,
  TableStyle,
  TableBodyStyle,
  ItemData,
  ItemDescriptionStyle,
  RatingTextStyles,
} from "./styles";

import ReactLoading from "react-loading";

import ReactStars from "react-rating-stars-component";

import { Button, Table } from "react-bootstrap";

import { formatMoney, withRouter } from "../../utils/misc";

class Menu extends Component {
  componentDidMount() {
    this.props.getMenu(this.props.router.params.id);
  }

  handleClick(item) {
    const { menu } = this.props;
    this.props.updateCart(
      menu.data.id,
      menu.data.name,
      { ...item, quantity: 0 },
      1
    );
  }

  getAverage(arr) {
    const count = arr?.reduce((acc, element) => acc + element.stars, 0);

    return count / arr?.length;
  }

  render() {
    const { menu } = this.props;
    const restaurant = menu.data;

    return (
      <>
        {menu.loading ? (
          <ReactLoading
            type={"spin"}
            style={{
              position: "absolute",
              width: "10vw",
              top: "40%",
              left: "0",
              right: "0",
              margin: "auto",
            }}
          />
        ) : (
          <PageStyle className="m-3">
            <h1>{restaurant.name}</h1>
            <div className="d-inline-flex justify-content-center align-items-center">
              <ReactStars
                count={5}
                isHalf={true}
                value={this.getAverage(restaurant.rates)}
                edit={false}
                size={50}
                activeColor="#ffd700"
              />
              <RatingTextStyles className="p-3 mt-2">
                ({this.getAverage(restaurant.rates)})
              </RatingTextStyles>
            </div>

            <Table borderless>
              <tbody>
                <tr className="mx-5">
                  <TableStyle>
                    {restaurant.menu?.options.map((element) => (
                      <td>
                        <TableBodyStyle>
                          <ItemPhoto className="m-2" photo={element.photo} />
                          <ItemData className="m-2">
                            <ItemDescriptionStyle>
                              <h4 style={{ fontWeight: "600" }}>
                                {element.name}
                              </h4>
                              {element.description}
                            </ItemDescriptionStyle>
                            <Button
                              style={{
                                width: "280px",
                                paddingBottom: 0,
                                borderRadius: "65px",
                              }}
                              variant="outline-danger"
                              type="button"
                              onClick={(e) => {
                                this.handleClick(element);
                              }}
                            >
                              <strong>
                                Adicionar item ao carrinho
                                <h4>{" +R$" + formatMoney(element.price)}</h4>
                              </strong>
                            </Button>
                          </ItemData>
                        </TableBodyStyle>
                      </td>
                    ))}
                  </TableStyle>
                </tr>
              </tbody>
            </Table>
          </PageStyle>
        )}
      </>
    );
  }
}

const mapStateToProps = ({ menu }) => ({ menu });

export default withRouter(
  connect(mapStateToProps, { ...MenuCreators, ...CartCreators })(Menu)
);
