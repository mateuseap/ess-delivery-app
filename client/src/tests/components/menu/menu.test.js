import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Menu from "../../../components/Menu";
import { BrowserRouter } from "react-router-dom";
import { INITIAL_STATE, Types } from "../../../store/ducks/menu";
import createSagaMiddleware from "redux-saga";
import { menuLoadedState } from "../dataUtils";

import React from "react";
import ReactStars from "react-rating-stars-component";
//React stars are being mocked due to a problem with Uniqueness in the snapshot render
jest.mock("react-rating-stars-component", () => () => <div>Hello World</div>);

import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";
import { ItemData } from "../../../components/Menu/styles";

const sagaMiddleware = createSagaMiddleware();

const mockStore = configureStore([sagaMiddleware]);

Enzyme.configure({ adapter: new Adapter() });

describe("Menu Component Integration Testing", () => {
  it("Renders loading correctly", () => {
    const store = mockStore({ menu: INITIAL_STATE });
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Menu />
        </BrowserRouter>
      </Provider>
    );

    const expectedActions = [{ type: Types.GET_MENU }];
    expect(store.getActions()).toEqual(expectedActions);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Renders data correctly", () => {
    const store = mockStore({
      menu: menuLoadedState,
    });
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Menu />
        </BrowserRouter>
      </Provider>
    );
    const expectedActions = [{ type: Types.GET_MENU }];
    expect(store.getActions()).toEqual(expectedActions);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
