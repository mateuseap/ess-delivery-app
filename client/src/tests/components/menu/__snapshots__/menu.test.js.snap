// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component Integration Testing Renders data correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ComponentWithRouterProp>
        <Connect(Menu)
          router={
            Object {
              "location": Object {
                "hash": "",
                "key": "default",
                "pathname": "/",
                "search": "",
                "state": null,
              },
              "navigate": [Function],
              "params": Object {},
            }
          }
        >
          <Menu
            cartError={[Function]}
            cartRequest={[Function]}
            cartSuccess={[Function]}
            getCart={[Function]}
            getMenu={[Function]}
            makeOrder={[Function]}
            menu={
              Object {
                "data": Object {
                  "addresses": Array [
                    "Avenida Caxangá, 634",
                  ],
                  "id": 2,
                  "menu": Object {
                    "destaqueIndex": 0,
                    "options": Array [
                      Object {
                        "description": "Hamburguer monstro de dar água na boca. ELIASLOL, O MELHOR COZINHEIRO DE LIMOEIRO!",
                        "item_id": 1,
                        "name": "Ogro Burguer com Batata Média",
                        "photo": "https://img.itdg.com.br/tdg/images/recipes/000/312/870/355695/355695_original.jpg",
                        "price": 25.5,
                      },
                      Object {
                        "description": "Coquinha geladinha HMMMMM!",
                        "item_id": 2,
                        "name": "Coca-cola lata",
                        "photo": "https://images-americanas.b2w.io/produtos/4575311740/imagens/coquinha-coca-cola-lata-220ml-desde-1886-sabor/4575311740_1_large.jpg",
                        "price": 5.5,
                      },
                    ],
                  },
                  "name": "Elias",
                  "photo": "https://media-cdn.tripadvisor.com/media/photo-s/05/85/e1/f5/sala-ristorante.jpg",
                  "rates": Array [
                    Object {
                      "feedback_text": "",
                      "stars": 4.5,
                      "user_id": "1",
                    },
                  ],
                },
                "error": false,
                "loading": false,
              }
            }
            menuError={[Function]}
            menuRequest={[Function]}
            menuSuccess={[Function]}
            router={
              Object {
                "location": Object {
                  "hash": "",
                  "key": "default",
                  "pathname": "/",
                  "search": "",
                  "state": null,
                },
                "navigate": [Function],
                "params": Object {},
              }
            }
            updateCart={[Function]}
          >
            <styled.div
              className="m-3"
            >
              <div
                className="sc-bczRLJ drhxhA m-3"
              >
                <h1>
                  Elias
                </h1>
                <div
                  className="d-inline-flex justify-content-center align-items-center"
                >
                  <Component
                    activeColor="#ffd700"
                    count={5}
                    edit={false}
                    isHalf={true}
                    size={50}
                    value={4.5}
                  >
                    <div>
                      Hello World
                    </div>
                  </Component>
                  <styled.h4
                    className="p-3 mt-2"
                  >
                    <h4
                      className="sc-gKXOVf kRfiiF p-3 mt-2"
                    >
                      (
                      4.5
                      )
                    </h4>
                  </styled.h4>
                </div>
                <ForwardRef
                  borderless={true}
                >
                  <table
                    className="table table-borderless"
                  >
                    <tbody>
                      <tr
                        className="mx-5"
                      >
                        <styled.div>
                          <div
                            className="sc-gsnTZi fLZfYw"
                          >
                            <td
                              key="1"
                            >
                              <styled.div
                                className="p-2"
                              >
                                <div
                                  className="sc-dkzDqf kZwptN p-2"
                                >
                                  <styled.img
                                    className="m-2"
                                    photo="https://img.itdg.com.br/tdg/images/recipes/000/312/870/355695/355695_original.jpg"
                                  >
                                    <img
                                      className="sc-hKMtZM kqihvI m-2"
                                    />
                                  </styled.img>
                                  <styled.div
                                    className="m-4"
                                  >
                                    <div
                                      className="sc-eCYdqJ bcsQqv m-4"
                                    >
                                      <styled.p>
                                        <p
                                          className="sc-jSMfEi bKJAuf"
                                        >
                                          <h4
                                            style={
                                              Object {
                                                "fontWeight": "600",
                                              }
                                            }
                                          >
                                            Ogro Burguer com Batata Média
                                          </h4>
                                          Hamburguer monstro de dar água na boca. ELIASLOL, O MELHOR COZINHEIRO DE LIMOEIRO!
                                        </p>
                                      </styled.p>
                                      <Button
                                        active={false}
                                        disabled={false}
                                        nome="ADICIONAR"
                                        onClick={[Function]}
                                        style={
                                          Object {
                                            "borderRadius": "65px",
                                            "paddingBottom": 0,
                                            "width": "280px",
                                          }
                                        }
                                        type="button"
                                        variant="outline-danger"
                                      >
                                        <button
                                          className="btn btn-outline-danger"
                                          disabled={false}
                                          nome="ADICIONAR"
                                          onClick={[Function]}
                                          style={
                                            Object {
                                              "borderRadius": "65px",
                                              "paddingBottom": 0,
                                              "width": "280px",
                                            }
                                          }
                                          type="button"
                                        >
                                          <strong>
                                            Adicionar item ao carrinho
                                            <h4>
                                              + R$ 25,50
                                            </h4>
                                          </strong>
                                        </button>
                                      </Button>
                                    </div>
                                  </styled.div>
                                </div>
                              </styled.div>
                            </td>
                            <td
                              key="2"
                            >
                              <styled.div
                                className="p-2"
                              >
                                <div
                                  className="sc-dkzDqf kZwptN p-2"
                                >
                                  <styled.img
                                    className="m-2"
                                    photo="https://images-americanas.b2w.io/produtos/4575311740/imagens/coquinha-coca-cola-lata-220ml-desde-1886-sabor/4575311740_1_large.jpg"
                                  >
                                    <img
                                      className="sc-hKMtZM hHaEKO m-2"
                                    />
                                  </styled.img>
                                  <styled.div
                                    className="m-4"
                                  >
                                    <div
                                      className="sc-eCYdqJ bcsQqv m-4"
                                    >
                                      <styled.p>
                                        <p
                                          className="sc-jSMfEi bKJAuf"
                                        >
                                          <h4
                                            style={
                                              Object {
                                                "fontWeight": "600",
                                              }
                                            }
                                          >
                                            Coca-cola lata
                                          </h4>
                                          Coquinha geladinha HMMMMM!
                                        </p>
                                      </styled.p>
                                      <Button
                                        active={false}
                                        disabled={false}
                                        nome="ADICIONAR"
                                        onClick={[Function]}
                                        style={
                                          Object {
                                            "borderRadius": "65px",
                                            "paddingBottom": 0,
                                            "width": "280px",
                                          }
                                        }
                                        type="button"
                                        variant="outline-danger"
                                      >
                                        <button
                                          className="btn btn-outline-danger"
                                          disabled={false}
                                          nome="ADICIONAR"
                                          onClick={[Function]}
                                          style={
                                            Object {
                                              "borderRadius": "65px",
                                              "paddingBottom": 0,
                                              "width": "280px",
                                            }
                                          }
                                          type="button"
                                        >
                                          <strong>
                                            Adicionar item ao carrinho
                                            <h4>
                                              + R$ 5,50
                                            </h4>
                                          </strong>
                                        </button>
                                      </Button>
                                    </div>
                                  </styled.div>
                                </div>
                              </styled.div>
                            </td>
                          </div>
                        </styled.div>
                      </tr>
                    </tbody>
                  </table>
                </ForwardRef>
              </div>
            </styled.div>
          </Menu>
        </Connect(Menu)>
      </ComponentWithRouterProp>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Menu Component Integration Testing Renders loading correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ComponentWithRouterProp>
        <Connect(Menu)
          router={
            Object {
              "location": Object {
                "hash": "",
                "key": "default",
                "pathname": "/",
                "search": "",
                "state": null,
              },
              "navigate": [Function],
              "params": Object {},
            }
          }
        >
          <Menu
            cartError={[Function]}
            cartRequest={[Function]}
            cartSuccess={[Function]}
            getCart={[Function]}
            getMenu={[Function]}
            makeOrder={[Function]}
            menu={
              Object {
                "data": Array [],
                "error": false,
                "loading": false,
              }
            }
            menuError={[Function]}
            menuRequest={[Function]}
            menuSuccess={[Function]}
            router={
              Object {
                "location": Object {
                  "hash": "",
                  "key": "default",
                  "pathname": "/",
                  "search": "",
                  "state": null,
                },
                "navigate": [Function],
                "params": Object {},
              }
            }
            updateCart={[Function]}
          >
            <styled.div
              className="m-3"
            >
              <div
                className="sc-bczRLJ drhxhA m-3"
              >
                <h1 />
                <div
                  className="d-inline-flex justify-content-center align-items-center"
                >
                  <Component
                    activeColor="#ffd700"
                    count={5}
                    edit={false}
                    isHalf={true}
                    size={50}
                    value={NaN}
                  >
                    <div>
                      Hello World
                    </div>
                  </Component>
                  <styled.h4
                    className="p-3 mt-2"
                  >
                    <h4
                      className="sc-gKXOVf kRfiiF p-3 mt-2"
                    >
                      (
                      NaN
                      )
                    </h4>
                  </styled.h4>
                </div>
                <ForwardRef
                  borderless={true}
                >
                  <table
                    className="table table-borderless"
                  >
                    <tbody>
                      <tr
                        className="mx-5"
                      >
                        <styled.div>
                          <div
                            className="sc-gsnTZi fLZfYw"
                          />
                        </styled.div>
                      </tr>
                    </tbody>
                  </table>
                </ForwardRef>
              </div>
            </styled.div>
          </Menu>
        </Connect(Menu)>
      </ComponentWithRouterProp>
    </Router>
  </BrowserRouter>
</Provider>
`;
